### PR TITLE
Terraform 0.12

### PR DESCRIPTION
**Pull request by seboudry**
Hi!
Here's a TF 0.12 compliant version of this module :)

### CODE REVIEW

**Comment by seboudry** [comment]
In order to the `pre-commit` test to pass, an upgrade of `lsstsqre/terraform-pre-commit` image with TF 0.12 is needed.
https://github.com/lsst-sqre/docker-terraform-pre-commit/blob/master/Dockerfile

**Comment by jhoblitt** [comment]
I've added an image with tf 0.12.6.  The tag is `lsstsqre/terraform-pre-commit:tf-0.12.6`

https://github.com/lsst-sqre/docker-terraform-pre-commit/pull/2

**Comment by jhoblitt** [comment]
@seboudry It looks like recent releases of `pre-commit-terraform` + tf 0.12.6 will still accept the 0.11 syntax without forcing a conversion?

**Comment by jhoblitt** [comment]
FYI -- I'm low bandwidth today as I'm at a conference.  What do you think of `tf-0.12/0.10.x` as a release branch name?

File: main.tf [hunk 1]
```diff
@@ -1,43 +1,43 @@
 resource "kubernetes_deployment" "tiller_deploy" {
-  metadata {
+  metadata { # metadata
```
Comment (jhoblitt) [inline main.tf]: This should be a comment on the closing brace.

**Review by jhoblitt — CHANGES_REQUESTED** [review]
This looks good except for the odd relocation of the closing brace comments.

File: main.tf (hunk 1 above)
Comment (seboudry) [reply main.tf]: Didn't see that. It's probably a `0.12upgrade` command impact.
But for me, we didn't need closing brace comments ;)

**Comment by seboudry** [comment]
> I've added an image with tf 0.12.6. The tag is `lsstsqre/terraform-pre-commit:tf-0.12.6`
> 
> [lsst-sqre/docker-terraform-pre-commit#2](https://github.com/lsst-sqre/docker-terraform-pre-commit/pull/2)

Nice, the test passes with this new image. I'll push a new commit.

> @seboudry It looks like recent releases of `pre-commit-terraform` + tf 0.12.6 will still accept the 0.11 syntax without forcing a conversion?

I've tested `lsstsqre/terraform-pre-commit:tf-0.12.6` on current `0.10.x` branch:
- `pre-commit run -a` OK after auto modify `main.tf` by aligning closing brace comments (I'll remove them as they don't add value)

> FYI -- I'm low bandwidth today as I'm at a conference. What do you think of `tf-0.12/0.10.x` as a release branch name?

Why not have a `0.12.x` branch name? Did you follow TF versions on your versioning?

**Comment by seboudry** [comment]
Hi @jhoblitt!
Sorry to bother you, any update on this PR?
Have a nice day

**Comment by seboudry** [comment]
Hi @jhoblitt !
Any change for that PR to be merged? ;)

**Comment by kouk** [comment]
@jhoblitt @afausti @athornton @digitizdat @rra is there any chance this will pass review and get merged? It would be great. kthxbye

**Review by jhoblitt — APPROVED** [review]

**Comment by jhoblitt** [comment]
Yes, we can merge this but it can't go into the master branch as it will break tf 0.11. Note that the existing branches in this repo are named after the provider version, not tf release. Also note that this mod isn't necessary with helm v3 / terraform-provider-helm >= 1.0.0.

**Comment by jhoblitt** [comment]
Merged into a branch named `tf-0.12/0.10.x`.

**Pull request closed without merging**